pkg/liqoctl/status/peer/peerinfo.go

```diff
@@ -0,0 +1,352 @@
+// checkPeerExistence checks if the given foreign cluster exists.
+// returns a slice of strings containing the names of the missing peers and a boolean indicating if all the given peers exists.
+func (pic *PeerInfoChecker) checkPeerExistence(fcList *discoveryv1alpha1.ForeignClusterList) []string {
+	var fcNames []string
+	notExistingPeerNames := []string{}
+	for i := range fcList.Items {
+		fcNames = append(fcNames, fcList.Items[i].Spec.ClusterIdentity.ClusterName)
+	}
+	for _, rcn := range pic.pio.RemoteClusterNames {
+		if !slice.ContainsString(fcNames, rcn) {
+			notExistingPeerNames = append(notExistingPeerNames, rcn)
+		}
+	}
+	return notExistingPeerNames
+}
+
+// addPeerSection adds a section about the peering generic info.
+func (pic *PeerInfoChecker) addPeerSection(rootSection output.Section, foreignCluster *discoveryv1alpha1.ForeignCluster) {
+	rootSection.AddEntry("Type", string(foreignCluster.Spec.PeeringType))
+	directionSection := rootSection.AddSection("Direction")
+	outgoingStatus := peeringconditionsutils.GetStatus(foreignCluster, discoveryv1alpha1.OutgoingPeeringCondition)
+	directionSection.AddEntry("Outgoing", string(outgoingStatus))
+	incomingStatus := peeringconditionsutils.GetStatus(foreignCluster, discoveryv1alpha1.IncomingPeeringCondition)
+	directionSection.AddEntry("Incoming", string(incomingStatus))
+}
+
+// addAuthSection adds a section about the authentication status.
+func (pic *PeerInfoChecker) addAuthSection(rootSection output.Section, foreignCluster *discoveryv1alpha1.ForeignCluster) {
+	authSection := rootSection.AddSection("Authentication")
+	authStatus := peeringconditionsutils.GetStatus(foreignCluster, discoveryv1alpha1.AuthenticationStatusCondition)
+	authSection.AddEntry("Status", string(authStatus))
+	if pic.o.Verbose {
+		authSection.AddEntry("Auth URL", foreignCluster.Spec.ForeignAuthURL)
+		if foreignCluster.Spec.ForeignProxyURL != "" {
+			authSection.AddEntry("Auth Proxy URL", foreignCluster.Spec.ForeignProxyURL)
+		}
+	}
+}
+
+// addNetworkSection adds a section about the network configuration.
+func (pic *PeerInfoChecker) addNetworkSection(ctx context.Context, rootSection output.Section,
+	foreignCluster *discoveryv1alpha1.ForeignCluster, localClusterName string) error {
+	networkSection := rootSection.AddSection("Network")
+	networkStatus := peeringconditionsutils.GetStatus(foreignCluster, discoveryv1alpha1.NetworkStatusCondition)
+	networkSection.AddEntry("Status", string(networkStatus))
+
+	if pic.o.Verbose {
+		networkConfigs, err := liqogetters.ListNetworkConfigByLabel(ctx, pic.o.CRClient,
+			foreignCluster.Status.TenantNamespace.Local, labels.NewSelector())
+		if err != nil {
+			return err
+		}
+
+		localNetworkConfigSection := networkSection.AddSection("Local CIDRs")
+		remoteNetworkConfigSection := networkSection.AddSection("Remote CIDRs")
+		var selectedSection output.Section
+		var remoteSectionMsg, remotePodCIDRMsg, remoteExternalCIDRMsg string
+		for i := range networkConfigs.Items {
```

Comment:
A warning should be probably issued if one or both network configs are not found.